src/android/FirebasePlugin.java

```diff
@@ -728,13 +728,24 @@ public void onVerificationCompleted(PhoneAuthCredential credential) {
 
                             JSONObject returnResults = new JSONObject();
                             try {
-                                String verificationId = getPrivateField(credential, "zzfc");
-                                String code = getPrivateField(credential, "zzfd");
-
-                                returnResults.put("verificationId", verificationId);
-                                returnResults.put("code", code);
+                                String verificationId = null;
+                                String code = null;
+								
+                                Field[] fields = credential.getClass().getDeclaredFields();
+                                for (Field field : fields) {
+                                    Class type = field.getType();
+                                    if(type == String.class){
+                                        String value = getPrivateField(credential, field);
+                                        if(value == null) continue;
+                                        if(value.length() > 100) verificationId = value;
+                                        else if(value.length() >= 4 && value.length() <= 6) code = value;
+                                    }
+                                }
+
+                                returnResults.put("verificationId", verificationId != null ? verificationId : false);
```

Comment:
@soumak77 Do you have any concerns about returning two different types in the result (string or boolean)?  I am an old school java guy so I don't like it, but I think JS handles it ok and if you are ok with it, I can be too.

Reply:
Nice catch.  No I don't like returning different types. I'd rather return an object with two different properties, such as "verified (boolean)" and "verficationId (string)". If verified is false, the id would be expected to be null.

Reply:
Also, I use typescript with this plugin, so mixing types is not acceptable to me.